src/Payum/Core/Registry/DynamicRegistry.php

```diff
@@ -64,7 +77,10 @@ public function getGateway($name)
      */
     public function getGateways()
     {
-        return $this->staticRegistry->getGateways();
+        return array_replace(
```

Comment:
the behavior of the getGateways method depends on was getGateway method is called. Let's improve getGateways in a separate PR. https://github.com/Payum/Payum/issues/531